framework/src/meshgenerators/BlockDeletionGenerator.C

```diff
@@ -21,19 +20,39 @@ BlockDeletionGenerator::validParams()
   InputParameters params = ElementDeletionGeneratorBase::validParams();
 
   params.addClassDescription(
-      "Mesh generator which removes elements with the specified subdomain ID");
-  params.addRequiredParam<SubdomainID>("block_id", "The block to be deleted");
+      "Mesh generator which removes elements from the specified subdomains");
+  params.addParam<std::vector<SubdomainName>>("block", "The list of blocks to be deleted");
+  params.addDeprecatedParam<SubdomainID>(
+      "block_id", "The block to be deleted.", "Use block instead");
 
   return params;
 }
 
 BlockDeletionGenerator::BlockDeletionGenerator(const InputParameters & parameters)
-  : ElementDeletionGeneratorBase(parameters), _block_id(getParam<SubdomainID>("block_id"))
+  : ElementDeletionGeneratorBase(parameters),
+    _block_names(getParam<std::vector<SubdomainName>>("block"))
 {
+  // Handle deprecated parameter
+  if (isParamValid("block_id"))
+    _block_ids.push_back(getParam<SubdomainID>("block_id"));
+  if (!isParamValid("block_id") && !isParamValid("block"))
+    mooseError("The list of blocks to be deleted should be provided.");
+  if (isParamValid("block_id") && isParamValid("block"))
+    mooseError("Please do not provide both block and block_id to the BlockDeletionGenerator.");
+}
+
+std::unique_ptr<MeshBase>
+BlockDeletionGenerator::generate()
+{
+  if (isParamValid("block"))
+    // Get the list of block ids from the block names
+    _block_ids = MooseMeshUtils::getSubdomainIDs(*_input, _block_names);
```

Comment:
After this, you should look through the IDs and see if any of them are INVALID_BLOCK_ID, which will be the case if the block with a given name/id was not found on the mesh. and if so, call paramError with "block" and the block name/id.

Reply:
You may also need to do an additional check, if the name is an ID if the ID actually exists on the mesh... I don't think this is checked within getSubdomainIDs. If you pass an ID of 4, it returns 4 whether or not 4 exists on the mesh.

Reply:
Something like this, I think:

```
  if (isParamValid("block"))
  {
    // Get the list of block ids from the block names
    _block_ids = MooseMeshUtils::getSubdomainIDs(*_input, _block_names);

    std::set<SubdomainID> mesh_blocks;
    _input->subdomain_ids(mesh_blocks);

    for (std::size_t i = 0; i < _block_ids.size(); ++i)
      if (_block_ids[i] == Moose::INVALID_BLOCK_ID // name provided didn't exist within mesh
          || !mesh_blocks.count(_block_ids[i]))    // ID provided didn't exist within mesh
        paramError("block", "The block '", _block_names[i], "' was not found within the mesh");
  }
```

Reply:
Param checking is a good idea here. Adding it, thx